packages/components/checkbox/src/Checkbox.stories.tsx

```diff
@@ -173,3 +176,99 @@ export const GroupImproved: StoryFn = () => {
     </FormField>
   )
 }
+
+export const CustomImplementation: StoryFn = () => {
+  const CustomCheckbox = ({ children, checked, ...others }: CheckboxProps) => {
+    const id = useId()
+    const { value } = others
+
+    return (
+      <Label
+        id={id}
+        htmlFor={value}
+        className={cx(
+          'flex flex-wrap shadow rounded-md gap-md p-lg max-w-sz-320',
+          checked ? 'bg-success/dim-4' : '',
+          'focus-within:ring-1 focus-within:ring-outline-high',
```

Comment:
I would remove the `ring-1` here. Guideline for spark focus is 
- `focus-visible` (keyboard nav only) set a ring-2.
- `focus-within` with a hover `ring-1` is only for inputs where the user can type.


Either remove it or replace it with `focus-visible:ring-2`